tests/CMakeLists.txt

```diff
@@ -3,12 +3,20 @@ set(BATCH_TESTS_SRCS flatmap.cpp input_output_file.cpp reduce_by_key.cpp
                      read_from_stdin.cpp )
 add_executable(batch_tests ${BATCH_TESTS_SRCS} test_driver.cpp)
 target_link_libraries(batch_tests ${PICO_RUNTIME_LIB})
+target_link_libraries(batch_tests Catch)
+add_test(NAME batch_unit_tests COMMAND batch_tests)
+add_test(NAME stream_unit_tests COMMAND /bin/sh ${CMAKE_CURRENT_SOURCE_DIR}/run_stream_tests.sh)
+
 
 set(STREAMING_TESTS_SRCS streaming_reduce_by_key.cpp )
 add_executable(stream_tests ${STREAMING_TESTS_SRCS} test_driver.cpp)
 target_link_libraries(stream_tests ${PICO_RUNTIME_LIB})
 
-add_subdirectory(testdata)
+add_custom_command(TARGET batch_tests
+        POST_BUILD
+        COMMAND /bin/sh ${CMAKE_CURRENT_SOURCE_DIR}/create_inputs_tests.sh
+        )
```

Comment:
This is dangerous: if `batch_test` target is not built, no input data is generated. Consider also that parallel building could induce a race in input data.

The clean solution would be to treat input data as a target, that would be a dependency for `batch_tests` and `batch_streams` targets.

Reply:
done